metashare/api/models.py

```diff
@@ -179,6 +180,43 @@ class Meta:
         ordering = ("name",)
         unique_together = (("repo_owner", "repo_name"),)
 
+    def save(self, *args, **kwargs):
+        if self.repo_id and not self.hook_secret:
+            user = self.get_a_matching_user()
+            if user:
+                gh.create_updates_hook_for_repo(user=user, repository=self)
+        return super().save(*args, **kwargs)
+
+    def get_a_matching_user(self):
+        github_repository = GitHubRepository.objects.filter(
+            repo_id=self.repo_id
+        ).first()
+
+        if github_repository:
+            return github_repository.user
+
+        return None
+
+    def refresh_commits(self, user):
+        from .jobs import refresh_commits_job
+
+        refresh_commits_job.delay(user=user, repository=self)
+
+    @transaction.atomic
+    def add_commits(self, commits, ref):
+        # TODO: this only works if the ref starts with this string:
```

Comment:
Let's add a fallback here, where if we can't find the branch we just refresh all commits?